### PR TITLE
feat: validate percentComplete value to be from 0 to 100

### PR DESCRIPTION
**Pull request by khalatevarun**
Fixes: 
#1216  

* Adds validation to percentCompleted param to accept value between 0 to 100
* Adds integration tests for the above case

### CODE REVIEW

File: test/integration/tasks.test.js [hunk 1]
```diff
@@ -632,7 +668,32 @@ describe("Tasks", function () {
...
+        endsOn: 1234,
+        startedOn: 4567,
+        status: "COMPLETED",
+        percentCompleted: 100,
+        participants: [],
+        assignee: appOwner.username,
+        completionAward: { [DINERO]: 3, [NEELAM]: 300 },
+        lossRate: { [DINERO]: 1 },
+        isNoteworthy: true,
+      };
```
Comment (heyrandhir) [inline test/integration/tasks.test.js:687]: can we move this to fixture.

File: controllers/tasks.js [hunk 2]
```diff
@@ -233,10 +233,8 @@ const updateTaskStatus = async (req, res, next) => {
     if (task.taskData.status === TASK_STATUS.VERIFIED || req.body.status === TASK_STATUS.MERGED)
       return res.boom.forbidden("Status cannot be updated. Please contact admin.");
 
-    if (task.taskData.status === TASK_STATUS.COMPLETED && req.body.percentCompleted < 100) {
-      if (req.body.status === TASK_STATUS.COMPLETED || !req.body.status) {
-        return res.boom.badRequest("Task percentCompleted can't updated as status is COMPLETED");
-      }
```
Comment (heyrandhir) [inline controllers/tasks.js:239]: why are we removing this condition !

File: controllers/tasks.js [hunk 3]
```diff
@@ -233,10 +233,8 @@ const updateTaskStatus = async (req, res, next) => {
...
+    if (task.taskData.status === TASK_STATUS.COMPLETED) {
+      return res.boom.badRequest("Task details can't be updated as it is marked as COMPLETED");
```
Comment (heyrandhir) [inline controllers/tasks.js:237]: we should only check this if the request body has status. Maybe we would like to change the `completionAward` or `startDate` or some other parameter. if in the request body we receive new status and the current status is completed then we shouldn't allow. Also please check with @ankushdharkar once that whether is there any state after completed. Maybe `released` might come after completed. In that case we must allow `released`. just check with Ankush once.

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: controllers/tasks.js (hunk 3 above)
Comment (khalatevarun) [reply controllers/tasks.js]: changes reverted as discussed

File: controllers/tasks.js (hunk 2 above)
Comment (khalatevarun) [reply controllers/tasks.js]: changes reverted as discussed

File: test/integration/tasks.test.js (hunk 1 above)
Comment (khalatevarun) [reply test/integration/tasks.test.js]: this test case has been reverted

**Review by heyrandhir — APPROVED** [review]

**Review by prakashchoudhary07 — APPROVED** [review]

**Pull request merged**